libopenage/renderer/camera/camera.cpp

```diff
@@ -113,6 +80,15 @@ void Camera::move_rel(Eigen::Vector3f direction, float delta) {
 	this->move_to(this->scene_pos + (direction * delta));
 }
 
+void Camera::move_rel(Eigen::Vector3f direction, float delta, CameraBoundaries camera_boundaries) {
+	auto new_pos = calc_look_at(this->scene_pos + (direction * delta));
+
+	new_pos[0] = std::clamp(new_pos[0], camera_boundaries.x_min, camera_boundaries.x_max);
+	new_pos[2] = std::clamp(new_pos[2], camera_boundaries.z_min, camera_boundaries.z_max);
```

Comment:
I think it might make sense to move the clamping to `move_to(..)` and pass through the boundaries object. Then we can use the same code for any camera movement, e.g. `look_at_scene(..)`